src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -1732,7 +1732,14 @@ fn report_diagnostic(
     tcx.struct_span_lint_hir(lint, hir_id, sp, |lint| {
         let mut diag = lint.build(msg);
 
-        let span = super::source_span_for_markdown_range(tcx, dox, link_range, &item.attrs);
+        let span =
+            super::source_span_for_markdown_range(tcx, dox, link_range, &item.attrs).map(|sp| {
+                if dox.bytes().nth(link_range.start) == Some(b'`') {
```

Comment:
It's not actually required that the backticks match - can you also check that the ending backtick is present?
```suggestion
                if dox.bytes().nth(link_range.start) == Some(b'`') && dox.bytes().nth(link_range.end) == Some(b'`') {
```

Reply:
I was wondering originally why it wasn't checking both the beginning and the end. Well, I guess that answers my question. :)